pandas/core/window/rolling.py

```diff
@@ -1778,15 +1776,16 @@ def _get_cov(X, Y):
     )
 
     def corr(self, other=None, pairwise=None, **kwargs):
-        if isinstance(self.window, BaseIndexer):
-            validate_baseindexer_support("corr")
-
         if other is None:
             other = self._selected_obj
             # only default unset
             pairwise = True if pairwise is None else pairwise
         other = self._shallow_copy(other)
-        window = self._get_window(other) if not self.is_freq_type else self.win_freq
+
+        if isinstance(self.window, BaseIndexer):
+            window = self.window
+        else:
+            window = self._get_window(other) if not self.is_freq_type else self.win_freq
```

Comment:
Fix for `corr`.

Reply:
can you add a comment here and on cov explaining what e are doing (generally not specific to BaseIndexer)

Reply:
@jreback Done. Please take a look if it's what you had in mind.